### PR TITLE
add header "unordered_map" to util/cuda_helpers.hpp

### PR DESCRIPTION
**Pull request by fwcore**
Prevent compilation error for g++-11

**Context:**
When compiling this repo in a docker `pytorch/pytorch`:
* ubuntu version 18.04
* gcc version 11 (installed from ppa:ubuntu-toolchain-r/test)
* nvcc/CUDA version 11.7

This line
https://github.com/PennyLaneAI/pennylane-lightning-gpu/blob/1efbe63881aeed132e3f38d31f5ad4bf26186722/pennylane_lightning_gpu/src/util/cuda_helpers.hpp#L589 fails to compile.

**Description of the Change:**

Add missing header `<unordered_map>` to util/cuda_helpers.hpp

**Benefits:**

The above file compiles.

**Possible Drawbacks:**

**Related GitHub Issues:**

### CODE REVIEW

**Comment by multiphaseCFD** [comment]
Hi @fwcore, Thanks for reporting this bug and the fix!

**Review by AmintorDusko — CHANGES_REQUESTED** [review]
Thank you for the fix. Could you please fill out the changelog file?
You can find it [here](https://github.com/PennyLaneAI/pennylane-lightning-gpu/blob/main/.github/CHANGELOG.md).
Don't forget to add your name to the contributors' list.

**Comment by fwcore** [comment]
> Hi @fwcore, Thanks for reporting this bug and the fix! Could you please share the compilation errors? Thanks

Here is the relevant error message:
```
/workspace/pennylane-lightning-gpu/pennylane_lightning_gpu/src/util/cuda_helpers.hpp(589): error: expected a ";"
/workspace/pennylane-lightning-gpu/pennylane_lightning_gpu/src/util/cuda_helpers.hpp(597): warning: parsing restarts here after previous syntax error
/workspace/pennylane-lightning-gpu/pennylane_lightning_gpu/src/util/cuda_helpers.hpp(605): error: identifier "pauli_map" is undefined
```

**Comment by fwcore** [comment]
Thanks for reviewing the PR. I have updated the CHANGELOG.md. Please review again. Thanks.

File: .github/CHANGELOG.md [hunk 1]
```diff
@@ -10,10 +10,15 @@
 
 ### Bug fixes
 
+* add header "unordered_map" to util/cuda_helpers.hpp [(#86)](https://github.com/PennyLaneAI/pennylane-lightning-gpu/pull/86)
```
Comment (multiphaseCFD) [inline .github/CHANGELOG.md:13]:
```suggestion
* Add header `unordered_map` to `util/cuda_helpers.hpp`.
 [(#86)](https://github.com/PennyLaneAI/pennylane-lightning-gpu/pull/86)
```

File: .github/CHANGELOG.md [hunk 2]
```diff
@@ -10,10 +10,15 @@
...
+
 ### Contributors
 
 This release contains contributions from (in alphabetical order):
 
+Feng Wang
+
+
```
Comment (multiphaseCFD) [inline .github/CHANGELOG.md:21]:
```suggestion
```

**Comment by multiphaseCFD** [comment]
> > Hi @fwcore, Thanks for reporting this bug and the fix! Could you please share the compilation errors? Thanks
> 
> Here is the relevant error message:
> 
> ```
> /workspace/pennylane-lightning-gpu/pennylane_lightning_gpu/src/util/cuda_helpers.hpp(589): error: expected a ";"
> /workspace/pennylane-lightning-gpu/pennylane_lightning_gpu/src/util/cuda_helpers.hpp(597): warning: parsing restarts here after previous syntax error
> /workspace/pennylane-lightning-gpu/pennylane_lightning_gpu/src/util/cuda_helpers.hpp(605): error: identifier "pauli_map" is undefined
> ```

Thanks @fwcore for sharing the information. Just some additional comments on typo of `changelog`.

**Comment by fwcore** [comment]
> > > Hi @fwcore, Thanks for reporting this bug and the fix! Could you please share the compilation errors? Thanks
> > 
> > 
> > Here is the relevant error message:
> > ```
> > /workspace/pennylane-lightning-gpu/pennylane_lightning_gpu/src/util/cuda_helpers.hpp(589): error: expected a ";"
> > /workspace/pennylane-lightning-gpu/pennylane_lightning_gpu/src/util/cuda_helpers.hpp(597): warning: parsing restarts here after previous syntax error
> > /workspace/pennylane-lightning-gpu/pennylane_lightning_gpu/src/util/cuda_helpers.hpp(605): error: identifier "pauli_map" is undefined
> > ```
> 
> Thanks @fwcore for sharing the information. Just some additional comments on typo of `changelog`.

Sorry for the late reply.

Thanks @multiphaseCFD for the suggestions. Please check again for the new commit. Thanks.

**Review by AmintorDusko — APPROVED** [review]
Thank you @fwcore for your nice work!

**Comment by fwcore** [comment]
Thank you all. Have a nice day.

**Pull request merged**